provider/resource_keycloak_user_roles.go

```diff
@@ -0,0 +1,253 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+	"strings"
+)
+
+func resourceKeycloakUserRoles() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKeycloakUserRolesCreate,
+		Read:   resourceKeycloakUserRolesRead,
+		Update: resourceKeycloakUserRolesUpdate,
+		Delete: resourceKeycloakUserRolesDelete,
+		// This resource can be imported using {{realm}}/{{userId}}.
+		Importer: &schema.ResourceImporter{
+			State: resourceKeycloakUserRolesImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"realm_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"user_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"role_ids": {
```

Comment:
Do I understand correctly that "role_ids" contains both realm role ids and client role ids?

Reply:
Yes. That's once again adopted from `keycloak_group_roles`.

Reply:
Yup, Keycloak doesn't actually care if its a realm or client role, so in this sense, the provider doesn't either.  So if you have multiple `keycloak_role` resources, you can add each `keycloak_role.*.id` to this set without having to worry about whether the roles are attached to realms or clients.